### PR TITLE
CXX-2480 convert docs.mongodb.com links to mongodb.com/docs

### PR DESCRIPTION
**Pull request by adriandole**

### CODE REVIEW

**Review by eramongodb — CHANGES_REQUESTED** [review]
Also update:

- `http://` -> `https://`
- "http://docs.mongodb.org" in [theme.toml](https://github.com/mongodb/mongo-cxx-driver/blob/6a1716e72c73403cbb9902627578258344d87a38/docs/themes/mongodb/theme.toml#L13).
- "https://www.mongodb.org/dochub" in [configuration.md](https://github.com/mongodb/mongo-cxx-driver/blob/6a1716e72c73403cbb9902627578258344d87a38/docs/content/mongocxx-v3/configuration.md?plain=1#L89). Note, some of these links appear to be broken and will need to be updated to point to their new equivalent pages.

May also want to update `https://mongodb.com` -> `https://www.mongodb.com` for consistency.

**Review by kevinAlbs — COMMENTED** [review]
Suggest applying the commits added in [this branch](https://github.com/mongodb/mongo-cxx-driver/compare/master...kevinAlbs:mongo-cxx-driver:myticket.CXX-2480?expand=1).

Running the script noted in [DRIVERS-2249](https://jira.mongodb.org/browse/DRIVERS-2249?focusedCommentId=4515052&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-4515052) resulted in additional changes. Running a [script](https://gist.github.com/kevinAlbs/a422461fd57d8f0debc3d88280803cc0) to search for broken links resulted in finding some broken links.

File: src/mongocxx/gridfs/bucket.hpp [hunk 1]
```diff
@@ -58,7 +58,7 @@ namespace gridfs {
 ///
 /// See also the method documentation for `mongocxx::database::gridfs_bucket()`.
 ///
-/// @see http://www.mongodb.org/display/DOCS/GridFS
+/// @see https://www.mongodb.org/display/DOCS/GridFS
```
Comment (eramongodb) [inline src/mongocxx/gridfs/bucket.hpp:61]:
```suggestion
/// @see https://www.mongodb.com/display/DOCS/GridFS
```
-> https://www.mongodb.com/docs/manual/core/gridfs/

**Review by eramongodb — APPROVED** [review]
One suggestion; otherwise, LGTM.

**Review by kevinAlbs — APPROVED** [review]

**Pull request merged**